www/js/lib/zimArchive.js

```diff
@@ -328,7 +351,7 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
      */
     ZIMArchive.prototype.getDirEntryByTitle = function(title) {
```

Comment:
I think this funciton is now misnamed. There is now a tangible difference between "titles" and "urls", yet this funciton is actually used for getting dirEntires by "url", not by title. I could re-name it to `getDirEntryByUrl` (which would require renaming code throughout the app where it is called - 16 results in 3 files).

Reply:
I think `getDirEntryByPath` might be a more accurate name (it searches on namespace + filename, e.g. `A/path/to/interesting_file` or `C/user_assets/mobile_display.css`). However, I'm torn on whether to do it as part of this PR or as a separate one.

**In favour of doing it in this PR**: this PR clearly separates "titles" (with their own pointer list) from "urls". Articles have titles, URLs (which include assets) may have titles, but more often than not don't. The fact that this function searches the url pointer list, but is named `getDirEntryByTitle`, is now very confusing . I nearly made a mistake, and was half-way to forcing it to use the new title/article pointer list, before remembering that it seaches on namespace + filename. So this PR sort of requires us to rename this function.

**In favour of doing it as a separate PR**: there are lots and lots of references to this function, and a number of variable names would need changing in disparate functions to rename this properly. We use "title" liberally throughout the code to refer to a url or a path. It would introduce lots of changes only tangentially relevant to this PR (including changing names of properites in the messages passed between Service Worker and `app.js`).

Reply:
I agree that there is some confusion in the code namings around these concepts. Some cleanup would be indeed very welcome.
About doing that in a separate PR or not, I understand the pros and cons. You can choose what seems better in your opinion.

Reply:
Thanks @mossroy. I've renamed the function in the latest commit, and have cleaned up as much as was viable to do in the scope of this PR in the functions that directly use this one. I've opened #710 for a more thorough cleanup of the many other instances of 'title' being used where 'url' or 'path[name]' would be clearer and avoid ambiguity.